<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		<title>The Golden Age of Genomics</title>
		<style>@media (prefers-color-scheme:dark){.markdown-body,[data-theme=dark]{color-scheme:dark;--color-prettylights-syntax-comment:#8b949e;--color-prettylights-syntax-constant:#79c0ff;--color-prettylights-syntax-entity:#d2a8ff;--color-prettylights-syntax-storage-modifier-import:#c9d1d9;--color-prettylights-syntax-entity-tag:#7ee787;--color-prettylights-syntax-keyword:#ff7b72;--color-prettylights-syntax-string:#a5d6ff;--color-prettylights-syntax-variable:#ffa657;--color-prettylights-syntax-brackethighlighter-unmatched:#f85149;--color-prettylights-syntax-invalid-illegal-text:#f0f6fc;--color-prettylights-syntax-invalid-illegal-bg:#8e1519;--color-prettylights-syntax-carriage-return-text:#f0f6fc;--color-prettylights-syntax-carriage-return-bg:#b62324;--color-prettylights-syntax-string-regexp:#7ee787;--color-prettylights-syntax-markup-list:#f2cc60;--color-prettylights-syntax-markup-heading:#1f6feb;--color-prettylights-syntax-markup-italic:#c9d1d9;--color-prettylights-syntax-markup-bold:#c9d1d9;--color-prettylights-syntax-markup-deleted-text:#ffdcd7;--color-prettylights-syntax-markup-deleted-bg:#67060c;--color-prettylights-syntax-markup-inserted-text:#aff5b4;--color-prettylights-syntax-markup-inserted-bg:#033a16;--color-prettylights-syntax-markup-changed-text:#ffdfb6;--color-prettylights-syntax-markup-changed-bg:#5a1e02;--color-prettylights-syntax-markup-ignored-text:#c9d1d9;--color-prettylights-syntax-markup-ignored-bg:#1158c7;--color-prettylights-syntax-meta-diff-range:#d2a8ff;--color-prettylights-syntax-brackethighlighter-angle:#8b949e;--color-prettylights-syntax-sublimelinter-gutter-mark:#484f58;--color-prettylights-syntax-constant-other-reference-link:#a5d6ff;--color-fg-default:#e6edf3;--color-fg-muted:#848d97;--color-fg-subtle:#6e7681;--color-canvas-default:#0d1117;--color-canvas-subtle:#161b22;--color-border-default:#30363d;--color-border-muted:#21262d;--color-neutral-muted:rgba(110,118,129,0.4);--color-accent-fg:#2f81f7;--color-accent-emphasis:#1f6feb;--color-success-fg:#3fb950;--color-success-emphasis:#238636;--color-attention-fg:#d29922;--color-attention-emphasis:#9e6a03;--color-attention-subtle:rgba(187,128,9,0.15);--color-danger-fg:#f85149;--color-danger-emphasis:#da3633;--color-done-fg:#a371f7;--color-done-emphasis:#8957e5}}@media (prefers-color-scheme:light){.markdown-body,[data-theme=light]{color-scheme:light;--color-prettylights-syntax-comment:#57606a;--color-prettylights-syntax-constant:#0550ae;--color-prettylights-syntax-entity:#6639ba;--color-prettylights-syntax-storage-modifier-import:#24292f;--color-prettylights-syntax-entity-tag:#116329;--color-prettylights-syntax-keyword:#cf222e;--color-prettylights-syntax-string:#0a3069;--color-prettylights-syntax-variable:#953800;--color-prettylights-syntax-brackethighlighter-unmatched:#82071e;--color-prettylights-syntax-invalid-illegal-text:#f6f8fa;--color-prettylights-syntax-invalid-illegal-bg:#82071e;--color-prettylights-syntax-carriage-return-text:#f6f8fa;--color-prettylights-syntax-carriage-return-bg:#cf222e;--color-prettylights-syntax-string-regexp:#116329;--color-prettylights-syntax-markup-list:#3b2300;--color-prettylights-syntax-markup-heading:#0550ae;--color-prettylights-syntax-markup-italic:#24292f;--color-prettylights-syntax-markup-bold:#24292f;--color-prettylights-syntax-markup-deleted-text:#82071e;--color-prettylights-syntax-markup-deleted-bg:#ffebe9;--color-prettylights-syntax-markup-inserted-text:#116329;--color-prettylights-syntax-markup-inserted-bg:#dafbe1;--color-prettylights-syntax-markup-changed-text:#953800;--color-prettylights-syntax-markup-changed-bg:#ffd8b5;--color-prettylights-syntax-markup-ignored-text:#eaeef2;--color-prettylights-syntax-markup-ignored-bg:#0550ae;--color-prettylights-syntax-meta-diff-range:#8250df;--color-prettylights-syntax-brackethighlighter-angle:#57606a;--color-prettylights-syntax-sublimelinter-gutter-mark:#8c959f;--color-prettylights-syntax-constant-other-reference-link:#0a3069;--color-fg-default:#1F2328;--color-fg-muted:#656d76;--color-fg-subtle:#6e7781;--color-canvas-default:#ffffff;--color-canvas-subtle:#f6f8fa;--color-border-default:#d0d7de;--color-border-muted:hsla(210,18%,87%,1);--color-neutral-muted:rgba(175,184,193,0.2);--color-accent-fg:#0969da;--color-accent-emphasis:#0969da;--color-success-fg:#1a7f37;--color-success-emphasis:#1f883d;--color-attention-fg:#9a6700;--color-attention-emphasis:#9a6700;--color-attention-subtle:#fff8c5;--color-danger-fg:#d1242f;--color-danger-emphasis:#cf222e;--color-done-fg:#8250df;--color-done-emphasis:#8250df}}.markdown-body{-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%;margin:0;color:var(--color-fg-default);background-color:var(--color-canvas-default);font-family:-apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";font-size:16px;line-height:1.5;word-wrap:break-word}.markdown-body .octicon{display:inline-block;fill:currentColor;vertical-align:text-bottom}.markdown-body h1:hover .anchor .octicon-link:before,.markdown-body h2:hover .anchor .octicon-link:before,.markdown-body h3:hover .anchor .octicon-link:before,.markdown-body h4:hover .anchor .octicon-link:before,.markdown-body h5:hover .anchor .octicon-link:before,.markdown-body h6:hover .anchor .octicon-link:before{width:16px;height:16px;content:' ';display:inline-block;background-color:currentColor;-webkit-mask-image:url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' version='1.1' aria-hidden='true'><path fill-rule='evenodd' d='M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z'></path></svg>");mask-image:url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' version='1.1' aria-hidden='true'><path fill-rule='evenodd' d='M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z'></path></svg>")}.markdown-body details,.markdown-body figcaption,.markdown-body figure{display:block}.markdown-body summary{display:list-item}.markdown-body [hidden]{display:none!important}.markdown-body a{background-color:transparent;color:var(--color-accent-fg);text-decoration:none}.markdown-body abbr[title]{border-bottom:none;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}.markdown-body b,.markdown-body strong{font-weight:var(--base-text-weight-semibold,600)}.markdown-body dfn{font-style:italic}.markdown-body h1{margin:.67em 0;font-weight:var(--base-text-weight-semibold,600);padding-bottom:.3em;font-size:2em;border-bottom:1px solid var(--color-border-muted)}.markdown-body mark{background-color:var(--color-attention-subtle);color:var(--color-fg-default)}.markdown-body small{font-size:90%}.markdown-body sub,.markdown-body sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}.markdown-body sub{bottom:-.25em}.markdown-body sup{top:-.5em}.markdown-body img{border-style:none;max-width:100%;box-sizing:content-box;background-color:var(--color-canvas-default)}.markdown-body code,.markdown-body kbd,.markdown-body pre,.markdown-body samp{font-family:monospace;font-size:1em}.markdown-body figure{margin:1em 40px}.markdown-body hr{box-sizing:content-box;overflow:hidden;background:0 0;border-bottom:1px solid var(--color-border-muted);height:.25em;padding:0;margin:24px 0;background-color:var(--color-border-default);border:0}.markdown-body input{font:inherit;margin:0;overflow:visible;font-family:inherit;font-size:inherit;line-height:inherit}.markdown-body [type=button],.markdown-body [type=reset],.markdown-body [type=submit]{-webkit-appearance:button;appearance:button}.markdown-body [type=checkbox],.markdown-body [type=radio]{box-sizing:border-box;padding:0}.markdown-body [type=number]::-webkit-inner-spin-button,.markdown-body [type=number]::-webkit-outer-spin-button{height:auto}.markdown-body [type=search]::-webkit-search-cancel-button,.markdown-body [type=search]::-webkit-search-decoration{-webkit-appearance:none;appearance:none}.markdown-body ::-webkit-input-placeholder{color:inherit;opacity:.54}.markdown-body ::-webkit-file-upload-button{-webkit-appearance:button;appearance:button;font:inherit}.markdown-body a:hover{text-decoration:underline}.markdown-body ::placeholder{color:var(--color-fg-subtle);opacity:1}.markdown-body hr::before{display:table;content:""}.markdown-body hr::after{display:table;clear:both;content:""}.markdown-body table{border-spacing:0;border-collapse:collapse;display:block;width:max-content;max-width:100%;overflow:auto}.markdown-body td,.markdown-body th{padding:0}.markdown-body details summary{cursor:pointer}.markdown-body details:not([open])>:not(summary){display:none!important}.markdown-body [role=button]:focus,.markdown-body a:focus,.markdown-body input[type=checkbox]:focus,.markdown-body input[type=radio]:focus{outline:2px solid var(--color-accent-fg);outline-offset:-2px;box-shadow:none}.markdown-body [role=button]:focus:not(:focus-visible),.markdown-body a:focus:not(:focus-visible),.markdown-body input[type=checkbox]:focus:not(:focus-visible),.markdown-body input[type=radio]:focus:not(:focus-visible){outline:solid 1px transparent}.markdown-body [role=button]:focus-visible,.markdown-body a:focus-visible,.markdown-body input[type=checkbox]:focus-visible,.markdown-body input[type=radio]:focus-visible{outline:2px solid var(--color-accent-fg);outline-offset:-2px;box-shadow:none}.markdown-body a:not([class]):focus,.markdown-body a:not([class]):focus-visible,.markdown-body input[type=checkbox]:focus,.markdown-body input[type=checkbox]:focus-visible,.markdown-body input[type=radio]:focus,.markdown-body input[type=radio]:focus-visible{outline-offset:0}.markdown-body kbd{display:inline-block;padding:3px 5px;font:11px ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace;line-height:10px;color:var(--color-fg-default);vertical-align:middle;background-color:var(--color-canvas-subtle);border:solid 1px var(--color-neutral-muted);border-bottom-color:var(--color-neutral-muted);border-radius:6px;box-shadow:inset 0 -1px 0 var(--color-neutral-muted)}.markdown-body h1,.markdown-body h2,.markdown-body h3,.markdown-body h4,.markdown-body h5,.markdown-body h6{margin-top:24px;margin-bottom:16px;font-weight:var(--base-text-weight-semibold,600);line-height:1.25}.markdown-body h2{font-weight:var(--base-text-weight-semibold,600);padding-bottom:.3em;font-size:1.5em;border-bottom:1px solid var(--color-border-muted)}.markdown-body h3{font-weight:var(--base-text-weight-semibold,600);font-size:1.25em}.markdown-body h4{font-weight:var(--base-text-weight-semibold,600);font-size:1em}.markdown-body h5{font-weight:var(--base-text-weight-semibold,600);font-size:.875em}.markdown-body h6{font-weight:var(--base-text-weight-semibold,600);font-size:.85em;color:var(--color-fg-muted)}.markdown-body p{margin-top:0;margin-bottom:10px}.markdown-body blockquote{margin:0;padding:0 1em;color:var(--color-fg-muted);border-left:.25em solid var(--color-border-default)}.markdown-body ol,.markdown-body ul{margin-top:0;margin-bottom:0;padding-left:2em}.markdown-body ol ol,.markdown-body ul ol{list-style-type:lower-roman}.markdown-body ol ol ol,.markdown-body ol ul ol,.markdown-body ul ol ol,.markdown-body ul ul ol{list-style-type:lower-alpha}.markdown-body dd{margin-left:0}.markdown-body code,.markdown-body samp,.markdown-body tt{font-family:ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace;font-size:12px}.markdown-body pre{margin-top:0;margin-bottom:0;font-family:ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace;font-size:12px;word-wrap:normal}.markdown-body .octicon{display:inline-block;overflow:visible!important;vertical-align:text-bottom;fill:currentColor}.markdown-body input::-webkit-inner-spin-button,.markdown-body input::-webkit-outer-spin-button{margin:0;-webkit-appearance:none;appearance:none}.markdown-body .mr-2{margin-right:var(--base-size-8,8px)!important}.markdown-body::before{display:table;content:""}.markdown-body::after{display:table;clear:both;content:""}.markdown-body>:first-child{margin-top:0!important}.markdown-body>:last-child{margin-bottom:0!important}.markdown-body a:not([href]){color:inherit;text-decoration:none}.markdown-body .absent{color:var(--color-danger-fg)}.markdown-body .anchor{float:left;padding-right:4px;margin-left:-20px;line-height:1}.markdown-body .anchor:focus{outline:0}.markdown-body blockquote,.markdown-body details,.markdown-body dl,.markdown-body ol,.markdown-body p,.markdown-body pre,.markdown-body table,.markdown-body ul{margin-top:0;margin-bottom:16px}.markdown-body blockquote>:first-child{margin-top:0}.markdown-body blockquote>:last-child{margin-bottom:0}.markdown-body h1 .octicon-link,.markdown-body h2 .octicon-link,.markdown-body h3 .octicon-link,.markdown-body h4 .octicon-link,.markdown-body h5 .octicon-link,.markdown-body h6 .octicon-link{color:var(--color-fg-default);vertical-align:middle;visibility:hidden}.markdown-body h1:hover .anchor,.markdown-body h2:hover .anchor,.markdown-body h3:hover .anchor,.markdown-body h4:hover .anchor,.markdown-body h5:hover .anchor,.markdown-body h6:hover .anchor{text-decoration:none}.markdown-body h1:hover .anchor .octicon-link,.markdown-body h2:hover .anchor .octicon-link,.markdown-body h3:hover .anchor .octicon-link,.markdown-body h4:hover .anchor .octicon-link,.markdown-body h5:hover .anchor .octicon-link,.markdown-body h6:hover .anchor .octicon-link{visibility:visible}.markdown-body h1 code,.markdown-body h1 tt,.markdown-body h2 code,.markdown-body h2 tt,.markdown-body h3 code,.markdown-body h3 tt,.markdown-body h4 code,.markdown-body h4 tt,.markdown-body h5 code,.markdown-body h5 tt,.markdown-body h6 code,.markdown-body h6 tt{padding:0 .2em;font-size:inherit}.markdown-body summary h1,.markdown-body summary h2,.markdown-body summary h3,.markdown-body summary h4,.markdown-body summary h5,.markdown-body summary h6{display:inline-block}.markdown-body summary h1 .anchor,.markdown-body summary h2 .anchor,.markdown-body summary h3 .anchor,.markdown-body summary h4 .anchor,.markdown-body summary h5 .anchor,.markdown-body summary h6 .anchor{margin-left:-40px}.markdown-body summary h1,.markdown-body summary h2{padding-bottom:0;border-bottom:0}.markdown-body ol.no-list,.markdown-body ul.no-list{padding:0;list-style-type:none}.markdown-body ol[type="a s"]{list-style-type:lower-alpha}.markdown-body ol[type="A s"]{list-style-type:upper-alpha}.markdown-body ol[type="i s"]{list-style-type:lower-roman}.markdown-body ol[type="I s"]{list-style-type:upper-roman}.markdown-body ol[type="1"]{list-style-type:decimal}.markdown-body div>ol:not([type]){list-style-type:decimal}.markdown-body ol ol,.markdown-body ol ul,.markdown-body ul ol,.markdown-body ul ul{margin-top:0;margin-bottom:0}.markdown-body li>p{margin-top:16px}.markdown-body li+li{margin-top:.25em}.markdown-body dl{padding:0}.markdown-body dl dt{padding:0;margin-top:16px;font-size:1em;font-style:italic;font-weight:var(--base-text-weight-semibold,600)}.markdown-body dl dd{padding:0 16px;margin-bottom:16px}.markdown-body table th{font-weight:var(--base-text-weight-semibold,600)}.markdown-body table td,.markdown-body table th{padding:6px 13px;border:1px solid var(--color-border-default)}.markdown-body table td>:last-child{margin-bottom:0}.markdown-body table tr{background-color:var(--color-canvas-default);border-top:1px solid var(--color-border-muted)}.markdown-body table tr:nth-child(2n){background-color:var(--color-canvas-subtle)}.markdown-body table img{background-color:transparent}.markdown-body img[align=right]{padding-left:20px}.markdown-body img[align=left]{padding-right:20px}.markdown-body .emoji{max-width:none;vertical-align:text-top;background-color:transparent}.markdown-body span.frame{display:block;overflow:hidden}.markdown-body span.frame>span{display:block;float:left;width:auto;padding:7px;margin:13px 0 0;overflow:hidden;border:1px solid var(--color-border-default)}.markdown-body span.frame span img{display:block;float:left}.markdown-body span.frame span span{display:block;padding:5px 0 0;clear:both;color:var(--color-fg-default)}.markdown-body span.align-center{display:block;overflow:hidden;clear:both}.markdown-body span.align-center>span{display:block;margin:13px auto 0;overflow:hidden;text-align:center}.markdown-body span.align-center span img{margin:0 auto;text-align:center}.markdown-body span.align-right{display:block;overflow:hidden;clear:both}.markdown-body span.align-right>span{display:block;margin:13px 0 0;overflow:hidden;text-align:right}.markdown-body span.align-right span img{margin:0;text-align:right}.markdown-body span.float-left{display:block;float:left;margin-right:13px;overflow:hidden}.markdown-body span.float-left span{margin:13px 0 0}.markdown-body span.float-right{display:block;float:right;margin-left:13px;overflow:hidden}.markdown-body span.float-right>span{display:block;margin:13px auto 0;overflow:hidden;text-align:right}.markdown-body code,.markdown-body tt{padding:.2em .4em;margin:0;font-size:85%;white-space:break-spaces;background-color:var(--color-neutral-muted);border-radius:6px}.markdown-body code br,.markdown-body tt br{display:none}.markdown-body del code{text-decoration:inherit}.markdown-body samp{font-size:85%}.markdown-body pre code{font-size:100%}.markdown-body pre>code{padding:0;margin:0;word-break:normal;white-space:pre;background:0 0;border:0}.markdown-body .highlight{margin-bottom:16px}.markdown-body .highlight pre{margin-bottom:0;word-break:normal}.markdown-body .highlight pre,.markdown-body pre{padding:16px;overflow:auto;font-size:85%;line-height:1.45;color:var(--color-fg-default);background-color:var(--color-canvas-subtle);border-radius:6px}.markdown-body pre code,.markdown-body pre tt{display:inline;max-width:auto;padding:0;margin:0;overflow:visible;line-height:inherit;word-wrap:normal;background-color:transparent;border:0}.markdown-body .csv-data td,.markdown-body .csv-data th{padding:5px;overflow:hidden;font-size:12px;line-height:1;text-align:left;white-space:nowrap}.markdown-body .csv-data .blob-num{padding:10px 8px 9px;text-align:right;background:var(--color-canvas-default);border:0}.markdown-body .csv-data tr{border-top:0}.markdown-body .csv-data th{font-weight:var(--base-text-weight-semibold,600);background:var(--color-canvas-subtle);border-top:0}.markdown-body [data-footnote-ref]::before{content:"["}.markdown-body [data-footnote-ref]::after{content:"]"}.markdown-body .footnotes{font-size:12px;color:var(--color-fg-muted);border-top:1px solid var(--color-border-default)}.markdown-body .footnotes ol{padding-left:16px}.markdown-body .footnotes ol ul{display:inline-block;padding-left:16px;margin-top:16px}.markdown-body .footnotes li{position:relative}.markdown-body .footnotes li:target::before{position:absolute;top:-8px;right:-8px;bottom:-8px;left:-24px;pointer-events:none;content:"";border:2px solid var(--color-accent-emphasis);border-radius:6px}.markdown-body .footnotes li:target{color:var(--color-fg-default)}.markdown-body .footnotes .data-footnote-backref g-emoji{font-family:monospace}.markdown-body .pl-c{color:var(--color-prettylights-syntax-comment)}.markdown-body .pl-c1,.markdown-body .pl-s .pl-v{color:var(--color-prettylights-syntax-constant)}.markdown-body .pl-e,.markdown-body .pl-en{color:var(--color-prettylights-syntax-entity)}.markdown-body .pl-s .pl-s1,.markdown-body .pl-smi{color:var(--color-prettylights-syntax-storage-modifier-import)}.markdown-body .pl-ent{color:var(--color-prettylights-syntax-entity-tag)}.markdown-body .pl-k{color:var(--color-prettylights-syntax-keyword)}.markdown-body .pl-pds,.markdown-body .pl-s,.markdown-body .pl-s .pl-pse .pl-s1,.markdown-body .pl-sr,.markdown-body .pl-sr .pl-cce,.markdown-body .pl-sr .pl-sra,.markdown-body .pl-sr .pl-sre{color:var(--color-prettylights-syntax-string)}.markdown-body .pl-smw,.markdown-body .pl-v{color:var(--color-prettylights-syntax-variable)}.markdown-body .pl-bu{color:var(--color-prettylights-syntax-brackethighlighter-unmatched)}.markdown-body .pl-ii{color:var(--color-prettylights-syntax-invalid-illegal-text);background-color:var(--color-prettylights-syntax-invalid-illegal-bg)}.markdown-body .pl-c2{color:var(--color-prettylights-syntax-carriage-return-text);background-color:var(--color-prettylights-syntax-carriage-return-bg)}.markdown-body .pl-sr .pl-cce{font-weight:700;color:var(--color-prettylights-syntax-string-regexp)}.markdown-body .pl-ml{color:var(--color-prettylights-syntax-markup-list)}.markdown-body .pl-mh,.markdown-body .pl-mh .pl-en,.markdown-body .pl-ms{font-weight:700;color:var(--color-prettylights-syntax-markup-heading)}.markdown-body .pl-mi{font-style:italic;color:var(--color-prettylights-syntax-markup-italic)}.markdown-body .pl-mb{font-weight:700;color:var(--color-prettylights-syntax-markup-bold)}.markdown-body .pl-md{color:var(--color-prettylights-syntax-markup-deleted-text);background-color:var(--color-prettylights-syntax-markup-deleted-bg)}.markdown-body .pl-mi1{color:var(--color-prettylights-syntax-markup-inserted-text);background-color:var(--color-prettylights-syntax-markup-inserted-bg)}.markdown-body .pl-mc{color:var(--color-prettylights-syntax-markup-changed-text);background-color:var(--color-prettylights-syntax-markup-changed-bg)}.markdown-body .pl-mi2{color:var(--color-prettylights-syntax-markup-ignored-text);background-color:var(--color-prettylights-syntax-markup-ignored-bg)}.markdown-body .pl-mdr{font-weight:700;color:var(--color-prettylights-syntax-meta-diff-range)}.markdown-body .pl-ba{color:var(--color-prettylights-syntax-brackethighlighter-angle)}.markdown-body .pl-sg{color:var(--color-prettylights-syntax-sublimelinter-gutter-mark)}.markdown-body .pl-corl{text-decoration:underline;color:var(--color-prettylights-syntax-constant-other-reference-link)}.markdown-body g-emoji{display:inline-block;min-width:1ch;font-family:"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";font-size:1em;font-style:normal!important;font-weight:var(--base-text-weight-normal,400);line-height:1;vertical-align:-.075em}.markdown-body g-emoji img{width:1em;height:1em}.markdown-body .task-list-item{list-style-type:none}.markdown-body .task-list-item label{font-weight:var(--base-text-weight-normal,400)}.markdown-body .task-list-item.enabled label{cursor:pointer}.markdown-body .task-list-item+.task-list-item{margin-top:4px}.markdown-body .task-list-item .handle{display:none}.markdown-body .task-list-item-checkbox{margin:0 .2em .25em -1.4em;vertical-align:middle}.markdown-body .contains-task-list:dir(rtl) .task-list-item-checkbox{margin:0 -1.6em .25em .2em}.markdown-body .contains-task-list{position:relative}.markdown-body .contains-task-list:focus-within .task-list-item-convert-container,.markdown-body .contains-task-list:hover .task-list-item-convert-container{display:block;width:auto;height:24px;overflow:visible;clip:auto}.markdown-body ::-webkit-calendar-picker-indicator{filter:invert(50%)}.markdown-body .markdown-alert{padding:var(--base-size-8) var(--base-size-16);margin-bottom:16px;color:inherit;border-left:.25em solid var(--color-border-default)}.markdown-body .markdown-alert>:first-child{margin-top:0}.markdown-body .markdown-alert>:last-child{margin-bottom:0}.markdown-body .markdown-alert .markdown-alert-title{display:flex;font-weight:var(--base-text-weight-medium,500);align-items:center;line-height:1}.markdown-body .markdown-alert.markdown-alert-note{border-left-color:var(--color-accent-emphasis)}.markdown-body .markdown-alert.markdown-alert-note .markdown-alert-title{color:var(--color-accent-fg)}.markdown-body .markdown-alert.markdown-alert-important{border-left-color:var(--color-done-emphasis)}.markdown-body .markdown-alert.markdown-alert-important .markdown-alert-title{color:var(--color-done-fg)}.markdown-body .markdown-alert.markdown-alert-warning{border-left-color:var(--color-attention-emphasis)}.markdown-body .markdown-alert.markdown-alert-warning .markdown-alert-title{color:var(--color-attention-fg)}.markdown-body .markdown-alert.markdown-alert-tip{border-left-color:var(--color-success-emphasis)}.markdown-body .markdown-alert.markdown-alert-tip .markdown-alert-title{color:var(--color-success-fg)}.markdown-body .markdown-alert.markdown-alert-caution{border-left-color:var(--color-danger-emphasis)}.markdown-body .markdown-alert.markdown-alert-caution .markdown-alert-title{color:var(--color-danger-fg)}</style>
		<style>
			body {
				margin: 0;
			}

			.markdown-body-content {
				box-sizing: border-box;
				min-width: 200px;
				max-width: 980px;
				margin: 0 auto;
				padding: 45px;
			}

			@media (max-width: 767px) {
				.markdown-body-content {
					padding: 15px;
				}
			}

			.markdown-body {
				--base-size-8: 8px;
				--base-size-16: 16px;
			}
		</style>
	</head>
	<body class="markdown-body">
		<article class="markdown-body-content"><h1>The Golden Age of Genomics</h1>
<p>The Human Genome Project is one of the greatest scientific accomplishments of all time. Over<br>
than a decade, an international team sequenced and analyzed human DNA, which propelled our<br>
understanding of biology. It also paved the way for new technologies that are now contributing to<br>
breakthroughs in cancer treatments [2], personalized medicine [3], virology [4], and even other<br>
fields like anthropology [5]. The study of genomes is called genomics, and, to learn more about it,<br>
I'll guide you through the main steps of a whole-genome sequencing project. You can follow along<br>
with this Python notebook to execute all the algorithms that I will use on your own computer. We will see:</p>
<ul>
<li>How to sequence DNA from genetic material,</li>
<li>How to assemble a genome from sequencing data, and</li>
<li>How to annotate a genome to explain its functions.</li>
</ul>
<h2>Sequencing</h2>
<p>The first step is sequencing, that is, reading the genetic material. DNA is a long molecule made of<br>
units called nucleotides, of which there are four types. They bind into complementary base pairs<br>
that store the genetic information necessary to build the RNA molecules and proteins that make us<br>
live [6]. That's why the Human Genome Project was launched: by having access to the entire genetic<br>
material of a human being, scientists would obtain a more complete picture of the biological<br>
processes carried out by these molecules. DNA is challenging to sequence because it is very small<br>
and very long [7], but multiple sequencing technologies have been developed over the years to read it.</p>
<p>Sanger sequencing, or first-generation sequencing, developed in the 70s, uses chain termination [8].<br>
It begins with a single-stranded DNA fragment to which we attach a primer, a short complementary<br>
sequence. An enzyme called DNA polymerase then elongates the fragment by adding complementary<br>
nucleotides. Some of them are actually modified nucleotides called dideoxynucleotides. When the<br>
enzyme happens to incorporate one, it terminates elongation and we end up with fragments of different<br>
lengths. They can be separated by gel or capillary electrophoresis, where fragments travel at<br>
different speeds depending on their lengths. We can resolve which nucleotides make up the sequence<br>
by using separate reactions or fluorescent labelling. This method can sequence fragments of at most<br>
1000 base pairs.</p>
<p>Second-generation sequencing, which became available in the mid 2000s, amplifies DNA into millions<br>
of fragments that get sequenced in parallel, drastically increasing throughput compared to Sanger<br>
sequencing and reducing costs [8]. There are multiple ways to do it. For instance, Illumina detects<br>
the emission of light as fluorescent nucleotides are incorporated into DNA strands while Ion Torrent<br>
detects pH changes during DNA synthesis. Second-generation technologies are limited to relatively<br>
short fragments, between around 50 and 300 base pairs.</p>
<p>Third-generation sequencing, developed since the late 2000s [8], directly sequences long DNA<br>
fragments without amplification. The two main technologies of this generation are Oxford nanopore,<br>
which detects an electrical current as DNA traverses a nanopore, and PacBio SMRT, which detects<br>
fluorescence during DNA synthesis. They can sequence fragments that have tens of thousands of base<br>
pairs. The error rate of third-generation sequencing devices used to be high, but their accuracy<br>
has improved a lot over the last years [9].</p>
<p>The Human Genome Project used Sanger sequencing because that's what was available at the<br>
time, but it's very slow and expensive. Second- and third-generation sequencing have largely replaced<br>
it in modern projects because they are much faster and cheaper. An important point to keep in mind is<br>
that none of these technologies are perfectly accurate: they misread some base pairs and, depending<br>
on how the samples are prepared, the ends of the sequences might contain erroneous data. Sequencing<br>
files like FASTQ can indicate the quality score of each sequenced base pair [10], and we can use<br>
these scores to reject low-quality reads and trim extremities where necessary [11].</p>
<p>Another problem you might have noticed is that sequencing technologies read a limited number of base<br>
pairs. Human chromosomes are much much longer than that, so it's impossible to sequence them in<br>
one go, even with the latest technologies.</p>
<h2>Assembly</h2>
<p>The second step is to assemble the genome from short DNA sequence reads. We have two options:</p>
<ul>
<li>stitch them together to reconstruct the genome - this is called de novo assembly, or</li>
<li>compare them against a reference genome if one is available - this is called reference-based assembly.</li>
</ul>
<h3>De Novo Assembly</h3>
<p>Let's begin with de novo assembly. We overlap the reads to find contiguous segments - those are<br>
called contigs - and put them together to form a scaffold, which is a draft of the complete genome.<br>
We then fill the gaps and fix the errors in the scaffold to obtain the final genome [12].</p>
<p>A naïve algorithm could compare each read to all other reads and identify matches to find the<br>
contigs. That scales terribly; modern sequencing technologies produce millions of reads, and it<br>
would take too much time to do all these pairwise alignments.</p>
<p>An improvement of this approach is the <strong>overlap-layout-consensus</strong> method [13]. We build an overlap<br>
graph where each node corresponds to a read and edges indicate overlaps. To reconstruct the original<br>
sequence, we find the shortest path that visits all nodes once - this is called layout. If you're<br>
familiar with graph theory, you'll recognize this as the traveling salesman problem, and there is no<br>
known algorithm that can efficiently solve it for arbitrary graphs [14]. We can find the shortest<br>
path by trying all possible combinations, but that's inefficient for large graphs. We instead<br>
approximate the solution with a heuristic, for example, a greedy search that picks the shortest edge<br>
linked to each node. It often works, but it's not guaranteed to find the shortest path. The last<br>
stage is consensus, where we resolve discrepancies between reads. Overlap-layout-consensus is more<br>
efficient than the naïve approach because it reduces the number of pairwise alignments, but the<br>
construction of the graph can take time, especially if we have a lot of reads.</p>
<p>Another algorithm more adapted to a large number of reads is the <strong>De Bruijn graph</strong> [13]. We build<br>
a graph similar to the one we had before, but instead of using complete reads, we cut them in pieces<br>
called K-mers that have a predetermined length. They are linked when they overlap by a single base<br>
pair, which makes the graph faster to build than with the other method. We then reconstruct the original<br>
sequence by finding the shortest path. This is more challenging because the K-mers contain less<br>
information, which makes conflicts more difficult to resolve. So there's a tradeoff between the two<br>
methods: overlap-layout-consensus is better for long reads, and De Bruijn graphs are better for short reads.</p>
<p>One problem with de novo approaches is their poor ability to resolve repetitions. There are multiple<br>
ways to overlap repeating reads, so the algorithms may fail to reconstruct the original sequence.<br>
Unfortunately, as large fraction of the human genome consists of repeats [31], so scientists often have to<br>
use tricks to resolve these repetitions, like using longer reads or isolating specific sequences.</p>
<h3>Reference-Based Assembly</h3>
<p>Let's now look at reference-based assembly, also called alignment. If we are assembling reads obtained from<br>
an organism for which the genome of a similar organism has already been sequenced, we can align our<br>
reads to this reference genome. That's more efficient than patching all the reads together because<br>
it requires fewer operations. It can be used to assemble human DNA because we have a reference<br>
genome, but it cannot be used with more poorly-studied organisms.</p>
<p>Finding a match for each read by naïvely searching an entire genome would be time-consuming, but we can<br>
convert the reference genome into a format that facilitates alignment. We would do something similar<br>
to search numbers. If you need to find a particular value in a list, you can check them all<br>
one by one, which can be slow, or sort the numbers and then perform a binary search.<br>
Sorting the numbers takes some time, but once that's done the search becomes much faster.</p>
<p>One way to organize the reference genome into a format that accelerates alignment is the Burrows-Wheeler transform. We list all<br>
the sequences that we can obtain by shifting the reference genome by different offsets and then sort<br>
these strings to obtain the transform. Since this rearrangement is based on lexicographic order, we<br>
can iteratively narrow the search down to sequences that match our reads and find the closest<br>
alignment efficiently. Of course, computing the transform takes some time, but once it's done,<br>
we can rapidly align our reads [15].</p>
<p>To recap, we can combine short reads into a full DNA sequence by using de novo or reference-based<br>
methods, depending on the sequencing technology that we use and the availability of reference<br>
genomes. In some cases, we can even combine these approaches to obtain better results [16]. But an<br>
assembled genome, by itself, is not very useful, it's just a bunch of letters. We need to make sense<br>
out of it.</p>
<h2>Annotation</h2>
<p>The third step of a genome project is annotation, that is, documenting a genome. There are two main<br>
things we can do:</p>
<ul>
<li>structural annotation identifies elements of interests like genes and</li>
<li>functional annotation explains what these elements do.</li>
</ul>
<h3>Structural Annotation</h3>
<p>Let's start with structural annotation. Our goal is to find regions in the genome that serve a<br>
particular purpose, like protein-coding genes. They get transcribed into messenger RNA and then<br>
translated into protein­s by ribosomes that decode groups of three nucleotides. These groups are<br>
called codons, and they correspond to specific amino acids. For instance, the codon ATG corresponds<br>
to methionine and also indicates the start of the translation. The codon CGC corresponds to arginine<br>
and the codon TGA stops the translation. This set of rules is called the genetic code. It is pretty<br>
similar across all lifeforms, but there exist some variations. For example, the mitochondria in<br>
human cells use a slightly modified version of the standard genetic code [32].</p>
<p>The oldest approach to identify genes is called ab initio, meaning that we use the genome<br>
itself without relying on external information. Protein-coding genes typically comprise a start and a<br>
stop codon to initiate and terminate translation, so we can look for sequences that follow this pattern.<br>
Since codons are made of three nucleotides and because DNA can be read in two directions, there are<br>
six possible ways to translate a DNA sequence.</p>
<p>For example, let's have a look at the first 300 nucleotides of the reference genome of Escherichia<br>
coli, a bacteria that lives in the intestine of warm-blooded animals. We can find 4 regions that contain a start<br>
and a stop codon, but that does not necessarily mean that they encode a protein. Indeed, when checking the<br>
reference annotation, we realize that only one of the 4 regions that we have found is actually a gene.<br>
The 3 others are false positives, and ab initio methods often end up with a lot of these. We can<br>
refine the search by using heuristics to avoid false positives, such as eliminating sequences that<br>
are too short and looking around for promoters, sequences that help initiate transcription. That will<br>
weed out some false positives, but even for the relatively simple genes of bacteria, none of<br>
these improvements will lead entirely accurate gene predictions. And the genes in eukaryotes like humans<br>
are even more complex and difficult to identify [17].</p>
<p>Another approach is to work backward by isolating a protein or a piece of RNA, deducing the DNA<br>
sequence that was used to make it, and then mapping this reconstructed sequence to the genome. This is an<br>
evidence-based method. Simple sequences might directly map to a specific locus in the genome, but in many cases,<br>
mapping is not as straightforward. For example, in mammals, a lot of genes contain introns,<br>
sequences that get discarded after transcription. The gene product does not tell us where these<br>
introns are, which makes our reconstructed sequences more difficult to align [18].</p>
<p>As more DNA sequences became available over time, a new type of structural annotation based on<br>
homology became more widespread. In short, we use sequences that we already know and try to align<br>
them to find similar sequences in a genome. For instance, we can identify a gene in one bacteria and then<br>
try to find a similar gene in another bacteria. Homology-based methods were not very useful before<br>
modern sequencing technologies, when few sequences were available, but now that we have large,<br>
annotated sequence databases, they have become more efficient [17].</p>
<h3>Functional Annotation</h3>
<p>Functional annotations explain what the elements that we have identified do. When scientists<br>
discover the function of a gene through experimental means, they can write an entry in a database to<br>
describe it in detail.</p>
<p>For instance, you can open the human reference genome the interactive genome viewer of the NCBI and search for the ABO gene,<br>
which contributes to determining blood groups in humans. If you click on that gene, you'll land on<br>
a page that provides functional annotations along with related elements, possible interactions with<br>
other genes, the expression of the gene in different tissues and more information. The page even links<br>
to research papers that discuss these points in detail [19].</p>
<p>I've focused on genes in this section, and, while they are essential components of the genome, they<br>
represent a small fraction of the DNA in many organisms. Around 1.5 % of human DNA is coding [20],<br>
meaning that it's used to make proteins. Other parts our genome are functional non-coding DNA because<br>
they don't encode proteins but have other well-identified functions. For example, centromeres are<br>
structural elements essential for the replication of chromosomes [21] and regulatory elements are<br>
regions around genes that control their expression [20]. Other regions of our genome have less<br>
clearly-identified functions. For example, transposable elements are sequences that can change<br>
position, and they make up around 45 % of the human genome [28]. Some have been shown to participate<br>
in gene regulation, others increase genetic diversity and contribute to evolution, but there remain<br>
a lot of transposable elements with unidentified functions [29].</p>
<p>There's no consensus on the proportion of the human genome that can be considered "functional". One<br>
way to estimate it is to find which sequences are under selective pressure, meaning that organisms<br>
tend to preserve these sequences over generations because they provide an evolutionary advantage.<br>
A team of researchers estimated that around 8.2 % of the human genome can be considered functional<br>
according to this definition [24]. Another approach is to find the proportion of biochemically<br>
active sequences. In 2012, the ENCODE project found that 80 % of the human genome can be considered<br>
"functional" in the sense that it's active [25]. Depending on how you define the word, the<br>
proportion of "functional" DNA in our genome differs.</p>
<p>To summarize, we use annotations to identify elements in a genome and explain their functions.<br>
There remain a lot of functions to uncover and the definition of "functional" DNA is not universal.</p>
<h2>Conclusion</h2>
<p>Let's recap: a whole-genome sequencing project aims at sequencing the genome of an organism and<br>
explaining how it works. It consists in:</p>
<ul>
<li>sequencing DNA, typically with second and third generation technologies nowadays,</li>
<li>assembling the sequenced reads into a genome with de novo or reference-based approaches, and</li>
<li>annotating the genome to identify elements and explain their functions.</li>
</ul>
<p>Genomics has immensely progressed over the last decades. Nowadays, sequencing technologies are much<br>
more efficient, analysis programs are faster, and databases contain more and more annotations. The<br>
Human Genome Project took more than 10 years of international effort to build one reference genome,<br>
but we can now sequence hundreds of genomes in much less time [27]. Technological innovation and<br>
scientific collaboration have led us to an era of rapid advances in genomics, which results in new<br>
medical treatments and improves our understanding of life.</p>
<h2>Bonus: Elaborating on Functional and Nonfunctional DNA</h2>
<blockquote>
<p>This section is not included in the video.</p>
</blockquote>
<p>-&gt; Elaborating on the discussion about functional DNA starting at 12:50</p>
<p>Less than 10 % of our genome has clearly recognized functions (encode proteins, control gene<br>
expression, protect the extremities of chromosomes, ...) while the rest does not have obvious<br>
functions. It is often called "junk DNA" (or, more neutrally, "nonfunctional DNA") because it<br>
appears not to be useful for the organism, but we often end up discovering that junk DNA actually<br>
has functions. They can be less critical than protein-coding genes, though.</p>
<ul>
<li>Introns were mostly seen as nonfunctional when they were discovered because they get discarded<br>
after transcription, but they can actually be important to determine how genes are expressed.</li>
<li>Transposable elements were seen as nonfunctional because they mostly act as selfish elements, a<br>
bit like parasites that are interested in their own replication and providing no benefit to the<br>
organism. But some TEs like LINE-1 have been shown to regulate genes. Others seem to promote<br>
genetic diversity, which could help species evolve over time (not directly useful for most organisms,<br>
but could still be seen as somewhat functional in long-term evolutionary adaptation). Other TEs<br>
are still considered nonfunctional, but maybe we don't just understand them well enough.</li>
</ul>
<p>DNA sequences are complex and don't neatly fit into "functional" and "nonfunctional" categories.<br>
Some are essential for survival and clearly fit into "functional", but it's more difficult to<br>
categorize sequences that are "somewhat useful". Evolutionary biologists and biochemists use<br>
different definitions to categorize sequences, which explains why they end up with such different<br>
estimates. The article <a href="https://www.pnas.org/doi/full/10.1073/pnas.1318948111" rel="nofollow">https://www.pnas.org/doi/full/10.1073/pnas.1318948111</a> summarizes them.</p>
<p>This situation reminds me of other semantic debates in science, such as the definition of the word<br>
"planet". Pluto was considered a planet for decades, but as astronomers discovered new objects in<br>
the solar system of a similar size, it became more logical to exclude it from the list of planets.<br>
Apparently, some astronomers are still beefing with that redefinition. Maybe one day the scientific<br>
community will settle on a clear definition of "functional DNA", but even if that happens I'm fairly<br>
certain that some people will disagree with it.</p>
<h2 id="references">References</h2>
<ul>
<li>[0] National Human Genome Research Institute. "Genomics". <a href="https://www.genome.gov/genetics-glossary/genomics" rel="nofollow">https://www.genome.gov/genetics-glossary/genomics</a></li>
<li>[1] International Human Genome Sequencing Consortium. "Finishing the euchromatic sequence of the human genome". Nature 431, 931–945 (2004). <a href="https://doi.org/10.1038/nature03001" rel="nofollow">https://doi.org/10.1038/nature03001</a></li>
<li>[2] Chakravarty, D., Solit, D.B. "Clinical cancer genomic profiling". Nat Rev Genet 22, 483–501 (2021). <a href="https://doi.org/10.1038/s41576-021-00338-8" rel="nofollow">https://doi.org/10.1038/s41576-021-00338-8</a></li>
<li>[3] Heinken, A., Hertel, J., Acharya, G. et al. "Genome-scale metabolic reconstruction of 7,302 human microorganisms for personalized medicine". Nat Biotechnol 41, 1320–1331 (2023). <a href="https://doi.org/10.1038/s41587-022-01628-0" rel="nofollow">https://doi.org/10.1038/s41587-022-01628-0</a></li>
<li>[4] Frampton, Dan et al. "Genomic characteristics and clinical effect of the emergent SARS-CoV-2 B.1.1.7 lineage in London, UK: a whole-genome sequencing and hospital-based cohort study". The Lancet Infectious Diseases, Volume 21, Issue 9, 1246 - 1256 <a href="https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(21)00170-5/fulltext" rel="nofollow">https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(21)00170-5/fulltext</a></li>
<li>[5] Yang, X., Li, H., Wang, B. et al. "Population dynamics in Iron Age Xinjiang inferred from ancient genomes of the Zhagunluke site". Archaeol Anthropol Sci 17, 76 (2025). <a href="https://doi.org/10.1007/s12520-025-02186-7" rel="nofollow">https://doi.org/10.1007/s12520-025-02186-7</a></li>
<li>[6] Alberts, Bruce et al. "The Structure and Function of DNA". Molecular Biology of the Cell. 4th edition. <a href="https://www.ncbi.nlm.nih.gov/books/NBK26821/" rel="nofollow">https://www.ncbi.nlm.nih.gov/books/NBK26821/</a></li>
<li>[7] National Center for Biotechnology Information (US). "Chromosome Map" <a href="https://www.ncbi.nlm.nih.gov/books/NBK22266/" rel="nofollow">https://www.ncbi.nlm.nih.gov/books/NBK22266/</a></li>
<li>[8] Kübra Eren , Nursema Taktakoğlu , Ibrahim Pirim. "DNA Sequencing Methods: From Past to Present". (2022) The Eurasian Journal of Medicine. <a href="https://www.eajm.org/Content/files/sayilar/224/8.pdf" rel="nofollow">https://www.eajm.org/Content/files/sayilar/224/8.pdf</a></li>
<li>[9] Wenger, A.M., Peluso, P., Rowell, W.J. et al. "Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome". Nat Biotechnol 37, 1155–1162 (2019). <a href="https://doi.org/10.1038/s41587-019-0217-9" rel="nofollow">https://doi.org/10.1038/s41587-019-0217-9</a></li>
<li>[10] National Center for Biotechnological Information. "File Format Guide". <a href="https://www.ncbi.nlm.nih.gov/sra/docs/submitformats/" rel="nofollow">https://www.ncbi.nlm.nih.gov/sra/docs/submitformats/</a></li>
<li>[11] Bolger, A. M., Lohse, M., &amp; Usadel, B. (2014). Trimmomatic: A flexible trimmer for Illumina Sequence Data. Bioinformatics, btu170. <a href="http://www.usadellab.org/cms/?page=trimmomatic" rel="nofollow">http://www.usadellab.org/cms/?page=trimmomatic</a></li>
<li>[12] Dominguez Del Angel V, Hjerde E, Sterck L et al. "Ten steps to get started in Genome Assembly and Annotation". F1000Research 2018, 7(ELIXIR):148 (<a href="https://doi.org/10.12688/f1000research.13598.1" rel="nofollow">https://doi.org/10.12688/f1000research.13598.1</a>)</li>
<li>[13] Gifford, David K. "Lecture 6 - Genome Assembly". MIT OpenCourseWare. <a href="https://ocw.mit.edu/courses/7-91j-foundations-of-computational-and-systems-biology-spring-2014/e885f0eb376ea6c2045eb9d8847f106f_MIT7_91JS14_Lecture6.pdf" rel="nofollow">https://ocw.mit.edu/courses/7-91j-foundations-of-computational-and-systems-biology-spring-2014/e885f0eb376ea6c2045eb9d8847f106f_MIT7_91JS14_Lecture6.pdf</a></li>
<li>[14] Kuo, Marc. "Algorithms for the Travelling Salesman Problem". Routific. <a href="https://www.routific.com/blog/travelling-salesman-problem" rel="nofollow">https://www.routific.com/blog/travelling-salesman-problem</a></li>
<li>[15] Heng Li, Richard Durbin, "Fast and accurate short read alignment with Burrows–Wheeler transform", Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/btp324" rel="nofollow">https://doi.org/10.1093/bioinformatics/btp324</a></li>
<li>[16] Lischer, H.E.L., Shimizu, K.K. "Reference-guided de novo assembly approach improves genome reconstruction for related species". <a href="https://doi.org/10.1186/s12859-017-1911-6" rel="nofollow">https://doi.org/10.1186/s12859-017-1911-6</a></li>
<li>[17] Ejigu GF, Jung J. Review on the Computational Genome Annotation of Sequences Obtained by Next-Generation Sequencing. Biology. 2020; 9(9):295. <a href="https://doi.org/10.3390/biology9090295" rel="nofollow">https://doi.org/10.3390/biology9090295</a></li>
<li>[18] Liang C, Mao L, Ware D, Stein L. "Evidence-based gene predictions in plant genomes". Genome Res. 2009 Oct;19(10):1912-23. doi: 10.1101/gr.088997.108. Epub 2009 Jun 18. PMID: 19541913; PMCID: PMC2765265.</li>
<li>[19] National Center for Biotechnological Information. "Genome Data Viewer". <a href="https://www.ncbi.nlm.nih.gov/gdv/" rel="nofollow">https://www.ncbi.nlm.nih.gov/gdv/</a></li>
<li>[20] Richard Boland, C. "Non-coding RNA: It’s Not Junk". Dig Dis Sci 62, 1107–1109 (2017). <a href="https://doi.org/10.1007/s10620-017-4506-1" rel="nofollow">https://doi.org/10.1007/s10620-017-4506-1</a></li>
<li>[21] Hayden KE, Strome ED, Merrett SL, Lee HR, Rudd MK, Willard HF. Sequences associated with centromere competency in the human genome. Mol Cell Biol. 2013 Feb;33(4):763-72. doi: 10.1128/MCB.01198-12. Epub 2012 Dec 10. PMID: 23230266; PMCID: PMC3571344.</li>
<li>[22] Jennifer Witek; Shamim S. Mohiuddin. "Biochemistry, Pseudogenes" <a href="https://www.ncbi.nlm.nih.gov/books/NBK549832/" rel="nofollow">https://www.ncbi.nlm.nih.gov/books/NBK549832/</a></li>
<li>[23] Kim YJ, Lee J, Han K. Transposable Elements: No More 'Junk DNA'. Genomics Inform. 2012 Dec;10(4):226-33. doi: 10.5808/GI.2012.10.4.226. Epub 2012 Dec 31. PMID: 23346034; PMCID: PMC3543922.</li>
<li>[24] Rands CM, Meader S, Ponting CP, Lunter G (2014) "8.2% of the Human Genome Is Constrained: Variation in Rates of Turnover across Functional Element Classes in the Human Lineage". PLoS Genet 10(7): e1004525. <a href="https://doi.org/10.1371/journal.pgen.1004525" rel="nofollow">https://doi.org/10.1371/journal.pgen.1004525</a></li>
<li>[25] ENCODE Project Consortium. An integrated encyclopedia of DNA elements in the human genome. Nature. 2012 Sep 6;489(7414):57-74. doi: 10.1038/nature11247. PMID: 22955616; PMCID: PMC3439153.</li>
<li>[26] Eddy, Sean R. "The C-value paradox, junk DNA and ENCODE". Current Biology <a href="https://www.cell.com/current-biology/fulltext/S0960-9822(12)01154-2" rel="nofollow">https://www.cell.com/current-biology/fulltext/S0960-9822(12)01154-2</a></li>
<li>[27] Liao, WW., Asri, M., Ebler, J. et al. A draft human pangenome reference. Nature 617, 312–324 (2023). <a href="https://doi.org/10.1038/s41586-023-05896-x" rel="nofollow">https://doi.org/10.1038/s41586-023-05896-x</a></li>
<li>[28] Margot Correa et al, "The Transposable Element Environment of Human Genes Differs According to Their Duplication Status and Essentiality", Genome Biology and Evolution, Volume 13, Issue 5, May 2021, evab062, <a href="https://doi.org/10.1093/gbe/evab062" rel="nofollow">https://doi.org/10.1093/gbe/evab062</a></li>
<li>[29] Mangoni, D., Simi, A., Lau, P. et al. LINE-1 regulates cortical development by acting as long non-coding RNAs. Nat Commun 14, 4974 (2023). <a href="https://doi.org/10.1038/s41467-023-40743" rel="nofollow">https://doi.org/10.1038/s41467-023-40743</a></li>
<li>[30] Pray, L. (2008) Transposons: The jumping genes. Nature Education 1(1):204 <a href="https://www.nature.com/scitable/topicpage/transposons-the-jumping-genes-518/" rel="nofollow">https://www.nature.com/scitable/topicpage/transposons-the-jumping-genes-518/</a></li>
<li>[31] Liao, X., Zhu, W., Zhou, J. et al. "Repetitive DNA sequence detection and its role in the human genome". Commun Biol 6, 954 (2023). <a href="https://doi.org/10.1038/s42003-023-05322-y" rel="nofollow">https://doi.org/10.1038/s42003-023-05322-y</a></li>
<li>[32] Andrzej (Anjay) Elzanowski, Jim Ostell. "The Genetic Codes". <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi" rel="nofollow">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a></li>
<li>[33] National Center for Biotechnological Information, "GenBank and WGS Statistics". <a href="https://www.ncbi.nlm.nih.gov/genbank/statistics/" rel="nofollow">https://www.ncbi.nlm.nih.gov/genbank/statistics/</a></li>
</ul>
<p>Articles related to the history of the Human Genome Project. Not used for the video but interesting<br>
if you want to know more about the HGP.</p>
<ul>
<li>How volunteers were recruited for the HGP: <a href="https://www.statnews.com/2024/07/09/human-genome-project-untold-story-how-single-volunteer-became-genetics-foundation/" rel="nofollow">https://www.statnews.com/2024/07/09/human-genome-project-untold-story-how-single-volunteer-became-genetics-foundation/</a></li>
<li>How stock prices tanked when the US and UK governments took a stance against gene patents: <a href="https://archive.nytimes.com/www.nytimes.com/learning/students/pop/articles/031500sci-human-genome.html" rel="nofollow">https://archive.nytimes.com/www.nytimes.com/learning/students/pop/articles/031500sci-human-genome.html</a></li>
</ul>
<p>If you still have trouble wrapping your head around how most of the human genome is apparently not<br>
selected during evolution: <a href="https://en.wikipedia.org/wiki/Nearly_neutral_theory_of_molecular_evolution" rel="nofollow">https://en.wikipedia.org/wiki/Nearly_neutral_theory_of_molecular_evolution</a></p></article>
	</body>
</html>
