

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Literature Review &mdash; stelaro 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=01f34227"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Benchmarks" href="benchmarks.html" />
    <link rel="prev" title="Developer Guide" href="developer_guide.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            stelaro
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="user_guide.html">User Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="developer_guide.html">Developer Guide</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Literature Review</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#classification">Classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gpu-acceleration">GPU Acceleration</a></li>
<li class="toctree-l2"><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="benchmarks.html">Benchmarks</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">stelaro</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Literature Review</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/literature_review.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="literature-review">
<h1>Literature Review<a class="headerlink" href="#literature-review" title="Link to this heading"></a></h1>
<p>Metagenomic tools analyze microbiomes from sequencing data, which makes them useful in biological
research to detect pathogens, study antimicrobial resistance, and predict illnesses, among others.
Although not yet used in clinical settings, many researchers have developed models that can process
metagenomic data with increasing efficacy <span id="id1">[]</span>.</p>
<p>Selective amplification (e.g. 16S, 18S, ITS) of specific regions of microbial genomes have been
widely used in metagenomics studies, but they introduce bias and omit elements during analysis
<span id="id2">[<a class="reference internal" href="#id4" title="Alison B.R. McIntyre, Rachid Ounit, Ebrahim Afshinnekoo, and others. Comprehensive benchmarking and ensemble approaches for metagenomic classifiers. Genome Biology, 18(1):182, 2017. URL: https://doi.org/10.1186/s13059-017-1299-7, doi:10.1186/s13059-017-1299-7.">3</a>]</span>. Shotgun sequencing is thus becoming a more reliable way to study microbiomes
for a variety of tasks:</p>
<ul class="simple">
<li><p>Classification</p></li>
<li><p>Abundance estimation</p></li>
<li><p>Identification</p></li>
</ul>
<section id="classification">
<h2>Classification<a class="headerlink" href="#classification" title="Link to this heading"></a></h2>
<p>Several methods can classify sequences of a sample into taxa:</p>
<ul class="simple">
<li><p><strong>Alignment</strong> of sequences</p></li>
<li><p><strong>Composition</strong> with k-mer analysis</p></li>
<li><p><strong>Phylogenetics</strong> using models of sequence evolution</p></li>
</ul>
</section>
<section id="gpu-acceleration">
<h2>GPU Acceleration<a class="headerlink" href="#gpu-acceleration" title="Link to this heading"></a></h2>
<ul class="simple">
<li><p>2012 homology search: <a class="reference external" href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0036060">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0036060</a></p></li>
<li><p>2012 analysis: <a class="reference external" href="https://link.springer.com/article/10.1186/1752-0509-6-S1-S16">https://link.springer.com/article/10.1186/1752-0509-6-S1-S16</a></p></li>
<li><p>2021 sequence assembly: <a class="reference external" href="https://dl.acm.org/doi/abs/10.1145/3458817.3476212">https://dl.acm.org/doi/abs/10.1145/3458817.3476212</a></p></li>
<li><p>2021 classification: <a class="reference external" href="https://dl.acm.org/doi/abs/10.1145/3472456.3472460">https://dl.acm.org/doi/abs/10.1145/3472456.3472460</a></p></li>
</ul>
</section>
<section id="references">
<h2>References<a class="headerlink" href="#references" title="Link to this heading"></a></h2>
<div class="docutils container" id="id3">
<div role="list" class="citation-list">
<div class="citation" id="id10" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>1<span class="fn-bracket">]</span></span>
<p>Alexander Fritz, Patrick Hofmann, Stefan Majda, Elisabeth Dahms, Helene Marie Draese, Till R Lesker, Peter Belmann, Matthew Z DeMaere, Aaron E Darling, Alexander Sczyrba, Andreas Bremges, and Alice C McHardy. Camisim: simulating metagenomes and microbial communities. <em>Microbiome</em>, 7(1):17, 2019. URL: <a class="reference external" href="https://doi.org/10.1186/s40168-019-0633-6">https://doi.org/10.1186/s40168-019-0633-6</a>, <a class="reference external" href="https://doi.org/10.1186/s40168-019-0633-6">doi:10.1186/s40168-019-0633-6</a>.</p>
</div>
<div class="citation" id="id11" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span>2<span class="fn-bracket">]</span></span>
<p>Dirk Hackenberger, Hamna Imtiaz, Amogelang R. Raphenya, Brian P. Alcock, Hendrik N. Poinar, Gerard D. Wright, and Andrew G. McArthur. Carpdm: cost-effective antibiotic resistome profiling of metagenomic samples using targeted enrichment. <em>bioRxiv</em>, 2024. URL: <a class="reference external" href="https://www.biorxiv.org/content/early/2024/03/30/2024.03.27.587061">https://www.biorxiv.org/content/early/2024/03/30/2024.03.27.587061</a>, <a class="reference external" href="https://arxiv.org/abs/https://www.biorxiv.org/content/early/2024/03/30/2024.03.27.587061.full.pdf">arXiv:https://www.biorxiv.org/content/early/2024/03/30/2024.03.27.587061.full.pdf</a>, <a class="reference external" href="https://doi.org/10.1101/2024.03.27.587061">doi:10.1101/2024.03.27.587061</a>.</p>
</div>
<div class="citation" id="id4" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id2">3</a><span class="fn-bracket">]</span></span>
<p>Alison B.R. McIntyre, Rachid Ounit, Ebrahim Afshinnekoo, and others. Comprehensive benchmarking and ensemble approaches for metagenomic classifiers. <em>Genome Biology</em>, 18(1):182, 2017. URL: <a class="reference external" href="https://doi.org/10.1186/s13059-017-1299-7">https://doi.org/10.1186/s13059-017-1299-7</a>, <a class="reference external" href="https://doi.org/10.1186/s13059-017-1299-7">doi:10.1186/s13059-017-1299-7</a>.</p>
</div>
</div>
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="developer_guide.html" class="btn btn-neutral float-left" title="Developer Guide" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="benchmarks.html" class="btn btn-neutral float-right" title="Benchmarks" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Vincent Therrien.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>